docs/content/development/adding-system-info-collectors.md

```diff
@@ -1,6 +1,105 @@
 ---
 title: "Adding System Info Collectors"
 date: 2020-06-09T11:03:42+03:00
-draft: true
+draft: false
+tags: ["contribute"]
+weight: 80
 ---
 
+## What's this?
+
+This guide will show you how to create a new _System Info Collector_ for the Infection Monkey. _System Info Collectors_ are modules which each Monkey runs, that collect specific information and sends it back to the Island as part of the System Info Telemetry.
+
+### Do I need a new System Info Controller?
+
+If all you want is to execute a shell command, then there's no need to add a new collector - just configure the required commands in the Monkey Island configuration in the PBA section! Also, if there is a relevant collector and you only need to add more information to it, expand the existing one. Otherwise, you must add a new Collector.
+
+## How to add a new System Info Collector
+
+### Monkey side
+
+#### Framework
+
+1. Create your new collector in the following directory: `monkey/infection_monkey/system_info/collectors` by first creating a new file with the name of your collector.
+2. In that file, create a class that inherits from the `SystemInfoCollector` class:
+
+```py
+from infection_monkey.system_info.system_info_collector import SystemInfoCollector
+
+class MyNewCollector(SystemInfoCollector):
+```
+
+3. Set the Collector name in the C'tor, like so:
+
+```py
+class MyNewCollector(SystemInfoCollector):
+    def __init__(self):
+        super(MyNewCollector, self).__init__(name="MyNewCollector")
+```
+
+#### Implementation
+
+Override the `collect` method with your own implementation. See the `EnvironmentCollector.py` Collector for reference. You can log during collection as well.
+
+#### Configuration
+
+Add the new collector to `infection_monkey/config.py`
+
+### Island side
+
+#### Configuration
+
+##### `definitions`
+
+You'll need to add your Collector to the `config_schema.py` file, under `definitions/system_info_collectors_classes/anyOf`, like so:
+
+```json
+"system_info_collectors_classes": {
+    "title": "System Information Collectors",
+    "type": "string",
+    "anyOf": [
+        {
+            "type": "string",
+            "enum": [
+                "EnvironmentCollector"
+            ],
+            "title": "Which Environment this machine is on (on prem/cloud)",
+            "attack_techniques": []
+        },
+        {                               <=================================
+            "type": "string",           <=================================
+            "enum": [                   <=================================
+                "MyNewCollector"        <=================================
+            ],                          <=================================
+            "title": "My new title",    <=================================
+            "attack_techniques": []     <=================================
+        },
+    ],
+},
+```
+
+##### `properties`
```

Comment:
Same as above; can we make it to look more clearly like a sub-heading?

Reply:
Fixed